heuristic/state.py

```diff
@@ -72,8 +113,8 @@ def get_violations_per_week(self, weeks, time_periods_in_week, competencies, emp
                             for t in time_periods_in_week[c, week])
                         for week in weeks]
 
-        contracted_hours = [sum(-min(0, self.soft_vars["deviation_contracted_hours"][e, j])
-                                for e in employees)
+        contracted_hours = [-sum(self.soft_vars["deviation_contracted_hours"].get((e, j), 0)
```

Comment:
I want free capacity to take a negative number, and violated contracted hours to take a positive number. Will this be correct @hkgrov ?

Reply:
Yes, this is correct then